src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/MigrateBookingStatusTest.kt

```diff
@@ -0,0 +1,321 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.integration
+
+import com.ninjasquad.springmockk.SpykBean
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.BookingStatus
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.MigrationJobType
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.ServiceName
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.integration.givens.`Given a User`
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.integration.givens.`Given an Offender`
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.BookingEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.BookingRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserEntity
+
+class MigrateBookingStatusTest : MigrationJobTestBase() {
+  @SpykBean
+  lateinit var realBookingRepository: BookingRepository
+
+  @Test
+  fun `Should not migrate CAS1 booking with 'cancelled' status and returns 202 response`() {
+    `Given a User` { userEntity, jwt ->
+      `Given an Offender` { offenderDetails, _ ->
+        val approvedPremises = ServiceName.approvedPremises
+        val booking = createCAS1CancelledBooking(userEntity)
+
+        assertBookingStatusIsNull(booking, approvedPremises)
+
+        migrationJobService.runMigrationJob(MigrationJobType.bookingStatus, 1)
+
+        assertBookingStatusIsNull(booking, approvedPremises)
+      }
+    }
+  }
+
+  @Test
+  fun `Should not migrate CAS3 booking with existing status 'arrived' with existing departure entity and returns 202 response`() {
```

Comment:
If a booking with the arrived status has a departure record (I'm not sure we expect this situation to happen), should it not be updated to departed after all?

Reply:
If booking already has status, then we dont want migrate the booking. We only pick up booking for migration when the status=null and service=temporary-accommodation.